specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,592 @@
+  /// send respective texture to the renderer, instead it notifies it
+  /// with internal buffer id, which is the identity of the shared buffer.
+  /// The shared buffer is 2D texture, IDXGIResource, format and will be
+  /// exposed through shared HANDLE or IUnknown type through ICoreWebView2StagingTexture.
+  /// Whenever the host has new texture to write, it should ask
+  /// reusable ICoreWebView2StagingTexture from the GetAvailableBuffer,
+  /// which returns ICoreWebView2StagingTexture.
+  /// If the GetAvailableBuffer returns an error, then the host calls the
+  /// CreateBuffer to allocate new shared buffer.
+  /// The API also registers created shared handle to the browser once it
+  /// created the resource.
+  HRESULT CreateBuffer(
+    [in] UINT32 width,
+    [in] UINT32 height,
+    [out, retval] ICoreWebView2StagingTexture** buffer);
+  /// GetAvailableBuffer can be called on any thread like SetBuffer.
+  HRESULT GetAvailableBuffer([out, retval] ICoreWebView2StagingTexture** buffer);
+  /// Remove texture buffer when the host removes the backed 2D texture.
+  /// The host can save the existing resources by deleting 2D textures
+  /// when it changes the frame sizes.
+  HRESULT RemoveBuffer([in] ICoreWebView2StagingTexture* buffer);
+  /// Indicates that the buffer is ready to present.
+  /// The buffer must be retrieved from the GetAvailableBuffer.
+  /// The host writes new texture to the local shared 2D texture of
+  /// the buffer id, which is created via CreateBuffer.
+  /// SetBuffer API can be called in any thread.
+  HRESULT SetBuffer([in] ICoreWebView2StagingTexture* buffer,
+    [in] ULONGLONG timestamp);
+  /// Render texture that is current set ICoreWebView2StagingTexture.
+  HRESULT Present();
+  /// Stop streaming of the current stream id.
+  /// API calls of Present, CreateBuffer will fail after this
+  /// with an error of COREWEBVIEW2_TEXTURE_STREAM_ERROR_NO_VIDEO_TRACK_STARTED.
+  /// The Javascript can restart the stream with getTextureStream.
+  HRESULT Stop();
+  /// Event handler for those that occur at the Renderer side, the example
+  /// are CreateBuffer, Present, or Stop.
+  HRESULT add_TextureError(
+      [in] ICoreWebView2StagingTextureStreamTextureErrorEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for texture error event.
+  HRESULT remove_TextureError([in] EventRegistrationToken token);
+  /// Updates d3d Device when it is updated by RenderAdapterLUIDChanged
+  /// event.
+  HRESULT UpdateD3DDevice([in] IUnknown* d3dDevice);
+  /// Event handler for receiving texture by Javascript.
+  /// There is no Start event for web texture. Whenever texture are sent
+  /// from the Javascript, the event is triggered.
+  HRESULT add_WebTextureReceived(
+      [in] ICoreWebView2StagingTextureStreamWebTextureReceivedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  HRESULT remove_WebTextureReceived([in] EventRegistrationToken token);

```

Comment:
For the remove_* methods just a simple reference documentation like should be above each. This one and the next one.

  /// Remove listener for the {event name} event.